### PR TITLE
Create issue templates and instructions for release testing

### PR DESCRIPTION
**Pull request by jashapiro**
I used github's features to create some issue templates.

Originally, I was just going to create the create an issue template for new releases, but it seemed easy enough to just make all the templates at once. 

I used the provided templates for features and bugs, with some modifications, especially for the bug report one.
I expect features to be the one we use most of the time (or no template), and it is the most general.

The release checklist is the one that should need the most review though:
- Did I describe the steps to test a release candidate sufficiently? 
- Should I include more detail about what happens after all the tests pass? (Instructions for actually creating a release?)

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+name: Bug report
```
Comment (jaclyn-taroni) [inline .github/ISSUE_TEMPLATE/bug_report.md:2]: This looks good 👍🏻

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+name: Feature request
```
Comment (jaclyn-taroni) [inline .github/ISSUE_TEMPLATE/feature_request.md:2]: This one looks good, too 👍🏻

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+about: Prepare for a new release version of scpcaTools
+title: Prepare for scpcaTools release vX.X.X
+labels: release
+assignees: ''
+
+---
+
+## Steps for a new release of `scpcaTools`
+
+- [ ] Are all dependencies/issues expected for this release are resolved? Add any that are not resolved as dependencies for this issue.
```
Comment (jaclyn-taroni) [inline .github/ISSUE_TEMPLATE/release-checklist.md:12]: Do you mean mark this issue (the release) as blocked by the unresolved issues in ZenHub?

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+- [ ] Did all automated tests and the Docker build in this repo pass in CI? (You may have to wait for the docker image to build. Check [Github actions](https://github.com/AlexsLemonade/scpcaTools/actions/workflows/build-docker.yaml) before proceeding to the next step
+- [ ] Test the new docker build in `scpca-nf`
+  - [ ] Create a branch in [scpca-nf](https://github.com/AlexsLemonade/scpca-nf/) for testing the latest Docker version from the `main` branch.
+  - [ ] Update [containers.config](https://github.com/AlexsLemonade/scpca-nf/blob/main/config/containers.config) with the `edge` version of the scpca-tools container: ```SCPCATOOLS_CONTAINER = 'ghcr.io/alexslemonade/scpca-tools:edge'``` and push the branch to github.
+  - [ ] If necessary, update the `scpca-nf` workflow to accommodate changes in the `scpcaTools` package  
+  - [ ] Perform a test run of the workflow with:
+```nextflow run AlexsLemonade/scpca-nf -r "<TESTBRANCH>" -profile batch```
+You may want to specify libraries or runs for testing.
```
Comment (jaclyn-taroni) [inline .github/ISSUE_TEMPLATE/release-checklist.md:20]: I believe your point here is: You might actually want to run something on some data. Is that correct?

**Review by jaclyn-taroni — COMMENTED** [review]
This looks pretty good! I had a couple questions about the release checklist, which I believe can be resolved with rewording things. 

Regarding this question:

> Should I include more detail about what happens after all the tests pass? (Instructions for actually creating a release?)

My inclination is yes. I don't expect that it will need to be lengthy. I think you could change the current section title to make it more explicitly about testing/preparing for a release and add a new section `Creating a release`

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 3 above)
Comment (jashapiro) [reply .github/ISSUE_TEMPLATE/release-checklist.md:12]: Yes, I will reword.

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 4 above)
Comment (jashapiro) [reply .github/ISSUE_TEMPLATE/release-checklist.md:20]: Sort of... There are default libraries in the workflow, but if there is a particular system to test, others might be good choices. I will reword.

**Comment by jashapiro** [comment]
I updated some wording and added release instructions. The package was failing tests yesterday because it seems to be failing to download and install some dependencies. Not fully sure what to do about that, and it usually seems to be transient. If it fails again I might look into it a bit more in the interests of avoiding similar problems in the future.

File: .github/ISSUE_TEMPLATE/release-checklist.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+assignees: ''
+
+---
+
+## Steps for a new release of `scpcaTools`
+
+### Preparing for the release and testing
+
+- [ ] Are all of the issues planned for this release resolved? If there are any issues that are unresolved, mark this issue as blocked by those on ZenHub.
+- [ ] Did all automated tests and the Docker build in this repo pass in CI? (You may have to wait for the docker image to build. Check [Github actions](https://github.com/AlexsLemonade/scpcaTools/actions/workflows/build-docker.yaml) before proceeding to the next step
```
Comment (jaclyn-taroni) [inline .github/ISSUE_TEMPLATE/release-checklist.md:15]:
```suggestion
- [ ] Did all automated tests and the Docker build in this repo pass in CI? (You may have to wait for the docker image to build. Check [Github actions](https://github.com/AlexsLemonade/scpcaTools/actions/workflows/build-docker.yaml) before proceeding to the next step)
```

Or maybe the `)` goes after "wait for the docker image to build."

**Review by jaclyn-taroni — APPROVED** [review]
Caught one missing ), but other than that LGTM!

File: .github/ISSUE_TEMPLATE/release-checklist.md (hunk 5 above)
Comment (jashapiro) [reply .github/ISSUE_TEMPLATE/release-checklist.md]: Just made it not a parenthetical.

**Comment by jashapiro** [comment]
Now that the tests are passing, I will merge this.

**Pull request merged**